### ISSUE TITLE
TA bookings without bookingOfficeCountryCode always report "Room Price Not Found" (checkout POST sends countryCode="None" -> HTTP 400)

### ISSUE THREAD

**Issue by C2Tech-sys** [jdeath/CheckRoyalCaribbeanPrice#107]
### Symptom

A booked, assigned cabin (not GTY) is reported **"Not For Sale" / "Room Price Not Found"** on every run, even though the same category is buyable on royalcaribbean.com. This appears to affect **travel-agent bookings** generally.

### Root cause

TA bookings can come back from the bookings API **without a `bookingOfficeCountryCode`** key. The chain:

1. `_build_checkout_url()` puts `booking.get("bookingOfficeCountryCode")` — a Python `None` — into the params dict.
2. `urlencode()` stringifies it to the literal text `country=None` in the generated URL.
3. `parse_provided_URL()` reads the string `"None"` back verbatim — its `params.get("country", ["USA"])` default only fires when the key is *absent*, not when it holds `"None"`.
4. `get_room_price_via_API()` POSTs `countryCode: "None"` to `/checkout/api/v1/rooms/checkout`, which rejects it with **HTTP 400 BAD_INPUT** (`must match pattern "^[A-Z]{3}$"`), and the run logs "Room Price Not Found".

The cruel part: the availability GET (`/room-selection/type-and-subtype`) *tolerates* the bad country value, so the cabin passes the availability check and then dies at pricing — which makes it look like a category/availability problem when it isn't.

### Live-API evidence (public checkout endpoint, no auth, package ST07E490, sail 2027-01-24, subtype I1)

| `countryCode` sent | Result |
|---|---|
| `"USA"` | 200, rooms priced |
| `"None"` (the bug) | 400 BAD_INPUT, pattern `^[A-Z]{3}$` |
| `null` / key absent / `"US"` | 400 BAD_INPUT |

(While bisecting: `sailDate` must be the dashed form `2027-01-24` — `20270124` also 400s. The URL builder already dashes it, so that's only a hazard for anyone constructing the POST directly.)

### Why this is distinct from #105

#105 fixes GTY category parsing. This bug hits **assigned cabins** (`guarantee: false`, stateroom number present) whenever the booking-office country is missing — the category parses fine and availability passes; only the checkout pricing call fails.

### Fix (one line)

Filter `None` values out of the params dict in `_build_checkout_url()` before `urlencode()`, so `parse_provided_URL()`'s existing defaults (`country` → `"USA"`, currency → `"USD"`) apply:

```python
params = {key: value for key, value in params.items() if value is not None}
```

This also stops any other absent field (`r0e`/`r0f`/currency) from leaking as the literal `"None"`. We're running this patch (plus a regression test that round-trips a booking with the key missing) on our fork and the affected cabin prices correctly again. Happy to send it as a PR if you'd like it.

**Comment by jdeath** [comment]
It could be same root cause as #105 . The GTY could be a coincidence. Please do a PR, as it helps with tracking and getting credit! Thanks for finding this

**Comment by C2Tech-sys** [comment]
PR opened: https://github.com/jdeath/CheckRoyalCaribbeanPrice/pull/108

On the #105 overlap — you may be right that they're cousins. Our failing cabin was `guarantee: false` with an assigned room, so the trigger here is the missing `bookingOfficeCountryCode` rather than GTY category parsing. But the mechanism generalizes: *any* booking that reaches `_build_checkout_url()` with a missing field leaks the literal string `"None"` into the URL the same way — a GTY with an unresolved category/subtype would leak `r0e`/`r0f` as `"None"` too. The filter in the PR covers that whole family, and it composes cleanly with #105 (we're running both together on our fork against live bookings).
